llmfoundry/data/finetuning/collator.py

```diff
@@ -487,38 +487,42 @@ def _process_and_batch_encoder_decoder(
             return_tensors='pt',
         )
         # We're still missing decoder_input_ids and decoder_attention_mask
-        batch['decoder_input_ids'] = torch.cat([
+        pad_id = self.tokenizer.pad_token_id
```

Comment:
nit: same here, you could probably just do some type casting/hint on batch?

Reply:
same, gonna leave this one